src/chart_types/partition_chart/layout/types/viewmodel_types.ts

```diff
@@ -159,7 +159,7 @@ export interface ShapeTreeNode extends TreeNode, SectorGeomSpecY {
   path: LegendPath;
   dataName: DataName;
   value: number;
-  parent: ArrayNode;
+  [MODEL_KEY]: ArrayNode;
```

Comment:
That seems fine to me. Are you trying to avoid breaking changes? I'd say this would be breaking.

One way to avoid a breaking change would be to keep the `parent` key but add the `@deprecated` flag. Replace all usage in kibana with the `MODEL_KEY` string. Then later we could change the name but it would be a cleaner breaking change when we remove `parent`.

```suggestion
  /** @deprecated use `MODEL_KEY` instead */
  parent: ArrayNode;
  [MODEL_KEY]: ArrayNode;
```

Reply:
Thanks Nick! The plan would be:

Full compatibility stage, no breaking API, eg. after merging this PR (`"whatever"` changed back to `"parent"`)
- we still call it `parent`
- everything still works and compliant with our API
- a new value appears in the API, `MODEL_KEY`

Deprecation stage: 
- we and users can start replace `.parent` with `[MODEL_KEY]`
- still compatible with the API as is; something gets a `@deprecated` tag, at least conceptually, as I think we can't have two identically named props at the same time, though haven't tried

Breaking change: 
- When nobody in at least Kibana uses `.parent` for the model, we make the breaking change of `const MODEL_KEY = 'model'`
- though it's a breaking change, it'll no longer impact known users due to our assist during deprecation, while other users simply need to deal with a changing API

I'm not sure if the proposed commit would still work, pls. indicate if that's the case, glad to try. At least from the diff, it's not clear what the deprecation line belongs to